src/app/components/Map/Map.tsx

```diff
@@ -17,61 +16,232 @@ if (typeof import.meta.env.VITE_MAPBOX_ACCESSKEY === 'string') {
+    const route = data.geometry.coordinates
+
+    const geojson = {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        type: 'LineString',
+        coordinates: route,
+      },
+    }
+
+    const pointData = {
+      type: 'FeatureCollection',
+      features: [
+        {
+          type: 'Feature',
+          properties: {},
+          geometry: {
+            type: 'Point',
+            coordinates: location1,
+          },
+        },
+      ],
+    }
+    map.current
+      ? map.current.fitBounds(
+          [
+            location1, // southwestern corner of the bounds
+            location2, // northeastern corner of the bounds
+          ],
+          { padding: { top: 100, bottom: 100, left: 100, right: 100 } }
+        )
+      : null
+
+    const endData = {
+      type: 'FeatureCollection',
+      features: [
+        {
+          type: 'Feature',
+          properties: {},
+          geometry: {
+            type: 'Point',
+            coordinates: location2,
+          },
+        },
+      ],
+    }
+
+    if (map.current?.getSource('route')) {
+      map.current.getSource('route').setData(geojson)
+    } else {
+      map.current?.addLayer({
+        id: 'route',
+        type: 'line',
+        source: {
+          type: 'geojson',
+          data: geojson,
+        },
+        layout: {
+          'line-join': 'round',
+          'line-cap': 'round',
+        },
+        paint: {
+          'line-color': '#2d8f43',
+          'line-width': 5,
+          'line-opacity': 0.75,
+        },
+      })
+    }
+    if (map.current?.getSource('point')) {
+      map.current.getSource('point').setData(pointData)
+    } else {
+      map.current?.addLayer({
+        id: 'point',
+        type: 'circle',
+        source: {
+          type: 'geojson',
+          data: {
+            type: 'FeatureCollection',
+            features: [
+              {
+                type: 'Feature',
+                properties: {},
+                geometry: {
+                  type: 'Point',
+                  coordinates: location1,
+                },
+              },
+            ],
+          },
+        },
+        paint: {
+          'circle-radius': 10,
+          'circle-color': '#3887be',
+        },
+      })
+    }
+
+    if (map.current?.getLayer('end')) {
+      map.current.getSource('end').setData(endData)
+    } else {
+      map.current?.addLayer({
+        id: 'end',
+        type: 'circle',
+        source: {
+          type: 'geojson',
+          data: {
+            type: 'FeatureCollection',
+            features: [
+              {
+                type: 'Feature',
+                properties: {},
+                geometry: {
+                  type: 'Point',
+                  coordinates: location2,
+                },
+              },
+            ],
+          },
+        },
+        paint: {
+          'circle-radius': 10,
+          'circle-color': '#f30',
+        },
+      })
+    }
+  }
```

Comment:
is it possible to define this function in its own file? would make your code much more readable